docs/DeveloperGuide.md

```diff
@@ -1,34 +1,845 @@
+
+-   `SearchCommand` initializes the `taskType` to check which tasks the search function to search from and `searchParam` to get the search query that the user inputs.
+
+-   Similar to the `SearchCommand`, `SearchdCommand` initializes `taskType` to check the tasks that the search function has to search through and `searchParam` to get the search query that the user inputs. It also has a `date` parameter to check the date that the users wants to search from
+
+Given below is an example usage of the `Search` command:  
+
+**Step 1**  
+The user launches the app and retrieves the tasks that are saved under a local file using `Storage`.
+
+**Step 2**  
+The user enters `search t\{TASK TYPE} n\{SEARCH QUERY}` into the command line. Method `parseCommand()` from the `Parser` class will be called to parse the command provided.
+
+**Step 3**  
+A new instance of `SearchCommand` with the `taskType`, `searchParam` and an ArrayList to store the index of search query , `storeIndex` will be initialized,
+
+-   If there are no tasks in the existing task list, it will initialize a new `CommandResult` class that prints out an error message, indicating an empty task list
+
+-   If there are tasks in the existing task list, it will call the `getSearchQueryAllTasks` or `getSearchQueryAssignments` or `getSearchQueryEvents`, according to the `taskType`.
+
+    -   In the `getSearchQuery` method, we will first get the updated task list from the `TaskList` class and parse through the task list to store results matching the search query into an ArrayList.
+
+        -   In the `getSearchQuery` method, we will first get the updated task list from the `getTaskArray` of the `TaskList` class and initialize a new ArrayList, `results` to store the results.
+
+        -   We will then loop though the updated task array to find matching tasks. Matching tasks will be added to the `results` ArrayList and the corresponding index will be stored to the `storeIndex` ArrayList.
+
+    -   Following that, the `getSearchQuery` method will call the `searchList` method to convert the stored results into a String format.
+
+    -   Lastly, the `searchList` method will call the `resultsList` method to return the search results.
+
+        -   If there are no matching search queries in the search results, `execute` method will then create a new `CommandResult` class to print an error message indicating an empty search results list.
+
+        -   If there are matching search queries in the search results, `execute` method will then create a new `CommandResults` class to print out the search results.
+
+Given below is an example usage of the `Searchd` command:  
+
+**Step 1**  
+The user launches the app and retrieves the tasks that are saved under a local file using `Storage`.
+
+**Step 2**  
+The user enters `searchd t\{TASK TYPE} n\{SEARCH QUERY} d\{DATE}` into the command line. Method `parseCommand()` from the `Parser` class will be called to parse the command provided.
+
+**Step 3**  
```

Comment:
Similar to the comment above, maybe can make it more concise to make this step easier to understand?